File: src/atcoder/newjudge2308/scala/newjudge2308_c.scala
import scala.io.StdIn
import scala.collection.immutable.TreeSet

def iotaSet(n: Int): TreeSet[Int] =
  if n < 0 then TreeSet[Int]() else (iotaSet(n - 1) + n)

def solve(bag: TreeSet[Int]): Unit =
  if bag.size > 0 then
    val ball = bag.last
    println(ball)
    solve(bag - ball - StdIn.readInt())

object Main extends App:
  val n = StdIn.readInt()
  val bag = iotaSet(n * 2 + 1)
  solve(bag)

File: src/atcoder/newjudge2308/clojure/newjudge2308_c.clj
(defn solve [bag]
  (if (< 0 (count bag))
    (let [ball (last bag)]
      (println ball)
      (solve (disj (disj bag ball) (read))))))

(def n (read))
(def balls (range (+ 2 (* 2 n))))
(def bag (apply sorted-set balls))
(solve bag)
